File: evidence/generated-systems/golang/manufacturing-on-golang/manufacturing-on-golang/internal/model/Operation.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Operation Declaration
//==============================================================
type Operation struct {
    gorm.Model
     OperationNumber                                    string
    Name                                    string
    SetupTime                                                            string
    StandardCycleTime                                                            string
    RoutingId         *uint
    Routing           *Routing `gorm:"foreignKey:RoutingId"`
    WorkCenterId         *uint
    WorkCenter           *WorkCenter `gorm:"foreignKey:WorkCenterId"`
    InspectionPlanId         *uint
    InspectionPlan           *InspectionPlan `gorm:"foreignKey:InspectionPlanId"`
    OperationType                      OperationType

// parent associations as their child

}

File: evidence/generated-systems/django/manufacturing-on-django/manufacturingOnDjango/models/Operation.py
from django.db import models
from manufacturingOnDjango.models.OperationType import OperationType

#======================================================================
# 
# Encapsulates data for model Operation
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Operation Declaration
#======================================================================
class Operation (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	operationNumber = models.CharField(max_length=200, null=True)
	name = models.CharField(max_length=200, null=True)
	setupTime = TimeDuration
	standardCycleTime = TimeDuration
	routing = models.ForeignKey('Routing', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	workCenter = models.ForeignKey('WorkCenter', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	inspectionPlan = models.ForeignKey('InspectionPlan', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	operationType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in OperationType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.operationNumber
		str = str + self.name
		str = str + self.setupTime
		str = str + self.standardCycleTime
		str = str + self.operationType
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Operation";
    
	def objectType(self):
		return "Operation";
